### PR TITLE
 Handle EOF HResult from async callback 

### PR DESCRIPTION
**Pull request by pakrym**
Fix for https://github.com/aspnet/AspNetCore/issues/6415

### CODE REVIEW

File: src/Servers/IIS/IIS/src/NativeMethods.cs [hunk 1]
```diff
@@ -14,7 +14,7 @@ internal static class NativeMethods
         internal const int ERROR_NOT_FOUND = unchecked((int)0x80070490);
         internal const int ERROR_OPERATION_ABORTED = unchecked((int)0x800703E3);
         internal const int ERROR_INVALID_PARAMETER = unchecked((int)0x80070057);
-        internal const int COR_E_IO = unchecked((int)0x80131620);
```
Comment (jkotalik) [inline src/Servers/IIS/IIS/src/NativeMethods.cs]: Do we remember what COR_E_IO was used for?

**Review by jkotalik — APPROVED** [review]

File: src/Servers/IIS/IIS/src/NativeMethods.cs (hunk 1 above)
Comment (pakrym) [reply src/Servers/IIS/IIS/src/NativeMethods.cs]: Nope

**Comment by pakrym** [comment]
Test failures are expected, I pushed the wrong HR to see if CI catches the regression.

**Comment by martincostello** [comment]
@pakrym Thanks for looking into this so promptly. Is this likely to meet the bar for inclusion in a 2.2.x release soon?

**Comment by pakrym** [comment]
It is pretty likely in my opinion.

**Pull request merged**